src/main/python/hydra/lib/runtestbase.py

```diff
@@ -221,6 +222,74 @@ def create_hydra_app(self, name, app_path, app_args, cpus, mem, ports=None, cons
         self.refresh_app_info(name)
         return r
 
+    def create_hydra_app_group(self, name, app_path, app_args, cpus, mem, apps_in_group,
+                               total_apps, ports=None, constraints=None):
+        """
+        Create an application in groups
+        @args:
+        name:                                      name of the app
+        app_path:                                  path to launch app from
+        app_args:                                  app args
+        cpus:                                      required cpus
+        mem:                                       required  mem
+        apps_in_group:                             number of apps to group together
+        total_apps:                                total apps to launch
```

Comment:
Total apps is what you want to ultimately scale to. Apps in group is the number of apps to be grouped together.    Right now only works for even number of apps in group and total apps. (will update if required)